--- a/drivers/net/phy/aquantia/aquantia_main.c
+++ b/drivers/net/phy/aquantia/aquantia_main.c
@@ -215,6 +215,10 @@ static int aqr_config_aneg(struct phy_de
 			      phydev->advertising))
 		reg |= MDIO_AN_VEND_PROV_5000BASET_FULL;
 
+	ret = phy_write_mmd(phydev, MDIO_MMD_PHYXS, 0xc441, 0x08);
+	if (ret < 0)
+		return ret;
+
 	ret = phy_modify_mmd_changed(phydev, MDIO_MMD_AN, MDIO_AN_VEND_PROV,
 				     MDIO_AN_VEND_PROV_1000BASET_HALF |
 				     MDIO_AN_VEND_PROV_1000BASET_FULL |
@@ -281,7 +285,7 @@ static int aqr_config_aneg_set_prot(stru
 
 static int aqr_config_intr(struct phy_device *phydev)
 {
-	bool en = phydev->interrupts == PHY_INTERRUPT_ENABLED;
+	bool en = true;//phydev->interrupts == PHY_INTERRUPT_ENABLED;
 	int err;
 
 	if (en) {
